libarchive/archive_parse_date.c

```diff
@@ -819,6 +819,23 @@ RelativeMonth(time_t Start, time_t Timezone, time_t RelMonth)
 		Timezone, DSTmaybe));
 }
 
+/*
+ * Parses and consumes an unsigned number.
+ * Returns 1 if any number is parsed. Otherwise, *value is unchanged.
+ */
+static char
+consume_unsigned_number(const char **in, time_t *value)
```

Comment:
I'm wondering what we should do here about possible overflows?

It might not be a problem:  An overflow here would "just" give us a nonsense date, which wouldn't be a serious problem for most uses.   Also, `time_t` doesn't have a standard associated `*_MAX` which makes overflow checking harder to set up.

Any thoughts?

Reply:
Good point... I'm not sure what the correct approach is. I do see that we seem to be moving to `la_int64_t` with the `__LA_TIME_T` macro which can make it easier, but it's indeed tricky with regular `time_t`. Now that I think of it, `time_t` isn't necessarily signed so `epoch = -epoch;` would be invalid. Furthermore, we also can't parse `@-1` because it's overloaded with the error sentinel 😞

We could naively detect the signedness of `time_t` and make an assumption of the limits based on `sizeof(time_t)` (not correct because`time_t` is allowed to have padding bytes).

Reply:
For the time being, I think we should remove support for negative @ timestamps since it may not be correct. Fortunately, I don't think most people need it anyways.

Reply:
Sounds reasonable.  I'll accept this as-is for now.

Someday, it would be nice to handle overflow here, but we can tackle that in a follow-on.